Add oauth2 examples for repo (#21)
streamnative/pulsar-examples: cloud/README.md

## How to get OAuth2 options

> When configuring and using OAuth2, we need to specify the following parameters:

Reviewer: change "we" into "you". Check all similar cases.
Use second person, for details, refer to https://developers.google.com/style/person

> - type

Reviewer 2: ```suggestion
- `type`
```

> - issuerUrl

Reviewer 2: ```suggestion
- `issuerUrl`
```

> - audience

Reviewer 2: ```suggestion
- `audience`
```

> - privateKey

Reviewer 2: ```suggestion
- `privateKey`
```

> - clientId

Reviewer 2: ```suggestion
- `clientId`
```

> For OAuth2 type, currently we only support `client_credentials`. So the value of the current type field is `client_credentials`

Reviewer 2: ```suggestion
For the OAuth2 `type` field, currently we only support `client_credentials`. 
```

> For privateKey, what we need to receive is the path of a private key data file. We can get this file in the following way:

Reviewer 2: ```suggestion
For the `privateKey` field, we need to get the path of a private key data file. The following example shows how to get this file:
```

> For the `privateKey` field, what we need to get the path of a private key data file. The following example shows how to get this file:

Reviewer 2: ```suggestion
For the `privateKey` field, we need to get the path of a private key data file. The following example shows how to get this file:
```

```shell script
$ snctl auth export-service-account [NAME] [flags]

Flags:
  -h, --help              help for export-service-account
  -f, --key-file string   Path to the private key file.
      --no-wait           Skip waiting for service account readiness.
```

Output:

```text
Wrote private key file <Path of your private key file>
```

> For `clientId` and `issuerUrl`, we can get the corresponding value from the privateKey file. For example:

Reviewer 2: ```suggestion
For the `clientId` and `issuerUrl` fields, we can get the corresponding value from the `privateKey` file. The following example shows how to get the `clientId` and `issueUrl` fields.
```

```text
{
"type":"sn_service_account",
"client_id":"0123456789abcdefg",
"client_secret":"ABCDEFG-JzAFKtj0Dcub9KF1WKN-qhFHBvgfAU_123456789-KI9",
"client_email":"test@gtest.auth0.com",
"issuer_url":"https://test.auth0.com"
}
```

> For `audience`, is the address of the accessed service.

Reviewer 2: ```suggestion
The `audience`field is the address of the accessed service.
```
